### PR TITLE
teleport-kube-agent-app: Secure Kubernetes access via Teleport

### PR DESCRIPTION
**Pull request by tuladhar**
Towards [#2712](https://github.com/giantswarm/roadmap/issues/2712)

This PR:

- Adds teleport-kube-agent app

### Testing

Description on how default-apps-cloud-director can be tested.

- [x] fresh install works
- [x] upgrade from previous version works

### Checklist

- [x] Update changelog in CHANGELOG.md, **check all commits are in it before release (renovate included)**.
- [x] Make sure `values.yaml` and `values.schema.json` are valid.

### CODE REVIEW

File: helm/default-apps-cloud-director/values.yaml [hunk 1]
```diff
@@ -156,6 +156,21 @@ apps:
     # used by renovate
     # repo: giantswarm/observability-bundle
     version: 0.8.7
+  teleportKubeAgent:
```
Comment (erkanerol) [inline helm/default-apps-cloud-director/values.yaml:159]: @tuladhar Does this require teleport binary in the nodes?

Comment (tuladhar) [reply helm/default-apps-cloud-director/values.yaml:159]: No, teleport binary is not required.

File: helm/default-apps-cloud-director/values.yaml [hunk 2]
```diff
@@ -156,6 +156,21 @@ apps:
...
+      configMap: true
+      secret: true
+    forceUpgrade: true
+    namespace: kube-system
+    # used by renovate
+    # repo: giantswarm/teleport-kube-agent-app
+    version: 0.6.0
+    extraConfigs:
+    - kind: configMap
+      name: "{{ $.Values.clusterName }}-teleport-kube-agent-config"
```
Comment (erkanerol) [inline helm/default-apps-cloud-director/values.yaml:173]: sorry. just came to my mind. who is supposed to create this cm?

Comment (erkanerol) [reply helm/default-apps-cloud-director/values.yaml:173]: If so, we shouldn't merge this PR until teleport-operator is a part of the vsphere-collection https://github.com/giantswarm/vsphere-app-collection/tree/main/flux-manifests 

Right?

Comment (tuladhar) [reply helm/default-apps-cloud-director/values.yaml:173]: teleport-operator is now part of vsphere-collection.
- https://github.com/giantswarm/vsphere-app-collection/commit/188e5d283dcebfb9ea77769d69acff4dc7a56860

**Review by kopiczko — APPROVED** [review]
LGTM

**Review by ssyno — APPROVED** [review]

File: helm/default-apps-cloud-director/values.yaml [hunk 3]
```diff
@@ -156,6 +156,21 @@ apps:
...
```
Comment (kopiczko) [inline helm/default-apps-cloud-director/values.yaml:173]: I think this is a recurring question. Maybe adding a comment that this is created with `giantswarm/teleport-operator` would be helpful

**Review by kopiczko — APPROVED** [review]

**Review by ssyno — APPROVED** [review]

**Comment by glitchcrab** [comment]
I would ignore that CI failure, every test passed except for the observability bundle deployment. Teleport was fine because the default-apps test passed:

```
  {"level":"info","ts":"2024-01-04T13:14:33Z","msg":"Checking if App status for t-kd96t1ycpvg4qqzedi-teleport-kube-agent is equal to 'deployed': deployed"}
```

**Review by glitchcrab — APPROVED** [review]

**Comment by erkanerol** [comment]
@glitchcrab @tuladhar Note that we have already a PM for the issue above https://github.com/giantswarm/giantswarm/issues/29287

**Pull request merged**